src/components/Reports/ReportFormAnnual.tsx

```diff
@@ -1,36 +1,389 @@
+    if (updateDetails) {
+      updateDetails('challenges', value);
+    }
+  });
+
+  const [keyLessons, setKeyLessons] = useState(report.annualDetails?.keyLessons ?? '');
+  useDebounce(keyLessons, DEBOUNCE_TIME_MS, (value) => {
+    if (updateDetails) {
+      updateDetails('keyLessons', value);
+    }
+  });
+
+  const [successStories, setSuccessStories] = useState(report.annualDetails?.successStories ?? '');
+  useDebounce(successStories, DEBOUNCE_TIME_MS, (value) => {
+    if (updateDetails) {
+      updateDetails('successStories', value);
+    }
+  });
+
+  const [isCatalytic, setIsCatalytic] = useState(report.annualDetails?.isCatalytic ?? false);
+  const [catalyticDetail, setCatalyticDetail] = useState(report.annualDetails?.catalyticDetail ?? '');
+  useDebounce(catalyticDetail, DEBOUNCE_TIME_MS, (value) => {
+    if (updateDetails) {
+      updateDetails('catalyticDetail', value);
+    }
+  });
+
+  const [opportunities, setOpportunities] = useState(report.annualDetails?.opportunities ?? '');
+  useDebounce(opportunities, DEBOUNCE_TIME_MS, (value) => {
+    if (updateDetails) {
+      updateDetails('opportunities', value);
+    }
+  });
+
+  const [nextSteps, setNextSteps] = useState(report.annualDetails?.nextSteps ?? '');
+  useDebounce(nextSteps, DEBOUNCE_TIME_MS, (value) => {
+    if (updateDetails) {
+      updateDetails('nextSteps', value);
+    }
+  });
+
+  const handleObservationMonthChange = (add: boolean, monthNumber: number) => {
+    const currentMonths = report.annualDetails?.bestMonthsForObservation ?? [];
+    if (updateDetails && add && !currentMonths.includes(monthNumber)) {
+      updateDetails('bestMonthsForObservation', [...currentMonths, monthNumber]);
+    } else if (updateDetails && !add && currentMonths.includes(monthNumber)) {
+      updateDetails(
+        'bestMonthsForObservation',
+        currentMonths.filter((m) => m !== monthNumber)
+      );
+    }
+  };
+
+  const handleSDGChange = (add: boolean, key: SustainableDevelopmentGoal) => {
+    if (updateDetails && add && !sdgList.includes(key)) {
+      const newSdg = [...sdgList];
+      newSdg.push(key);
+      setSdgList(newSdg);
+      updateDetails(
+        'sustainableDevelopmentGoals',
+        newSdg.map((s) => ({
+          goal: s,
+          progress: sdgProgressStates[SDG.findIndex((sdg) => s === sdg)],
+        }))
+      );
+    } else if (updateDetails && !add && sdgList.includes(key)) {
+      const newSdg = sdgList.filter((s) => s !== key);
+      setSdgList(newSdg);
+      updateDetails(
+        'sustainableDevelopmentGoals',
+        newSdg.map((s) => ({
+          goal: s,
+          progress: sdgProgressStates[SDG.findIndex((sdg) => s === sdg)],
+        }))
+      );
+    }
+  };
+
+  const handleIsCatalyticChange = (catalytic: boolean) => {
+    if (updateDetails) {
+      setIsCatalytic(catalytic);
+      updateDetails('isCatalytic', catalytic);
+    }
+  };
+
+  const handleFileUploadButtonClick = () => {
+    // TODO
```

Comment:
Will do this in a followup—this PR was getting big and I hadn't quite gotten this figured out